### PR TITLE
Big-endian fix: Base64Encoder

### PR DESCRIPTION
**Pull request by uweigand**
* Take byte order into accout when packing characters into an int

### CODE REVIEW

File: src/libraries/System.Memory/src/System/Buffers/Text/Base64Encoder.cs [hunk 1]
```diff
@@ -486,7 +486,14 @@ private static unsafe uint Encode(byte* threeBytes, ref byte encodingMap)
...
             uint i3 = Unsafe.Add(ref encodingMap, (IntPtr)(i & 0x3F));
 
-            return i0 | (i1 << 8) | (i2 << 16) | (i3 << 24);
+            if (BitConverter.IsLittleEndian)
+            {
+                return i0 | (i1 << 8) | (i2 << 16) | (i3 << 24);
+            }
+            else
+            {
+                return (i0 << 24) | (i1 << 16) | (i2 << 8) | i3;
```
Comment (gfoidl) [inline src/libraries/System.Memory/src/System/Buffers/Text/Base64Encoder.cs:495]: Is this actually needed?
In base64 the encoded data is stored as big endian like, and here it's written by https://github.com/dotnet/runtime/blob/44595c05ca033d1010397d05a55ab89bf078e56c/src/libraries/System.Memory/src/System/Buffers/Text/Base64Encoder.cs#L95 (~ `reinterpret_cast`) and not as `uint`. So I believe this change flips the order.

Comment (uweigand) [reply src/libraries/System.Memory/src/System/Buffers/Text/Base64Encoder.cs:495]: `Unsafe.WriteUnaligned` does always write in native byte order.  So on a big-endian system, the first byte written is the most-signficant byte of the `uint` etc.  This conflicts with the way the Encode routines currently prepare their result.  The intent is for `i0` to become the first byte written, `i1` the second byte written etc.  To achieve this effect, `i0` has be moved into the least-significant byte of the `uint` result on little-endian machines, but into the most-significant byte of the result on big-endian machines.

Note that this change actually fixes unit test case failures I saw on my big-endian machine.

Comment (gfoidl) [reply src/libraries/System.Memory/src/System/Buffers/Text/Base64Encoder.cs:495]: Ah, yes you're correct! Didn't think enough that it is written as `uint` / dword into memory. Thanks!

**Review by jkotas — APPROVED** [review]

**Pull request merged**